### PR TITLE
Implemented tests for functions in mode module

### LINKED ISSUE #921: Write Unit Tests for functions in mode module
The functions in game-state/mode.h currently do not have unit tests, so I will write unit tests for them.
Assignees: Peter
team:npc2

### PR DESCRIPTION
**Pull request by pkasner**
(Closes #921) 
The functions contained mode.h did not have any unit tests, so I have implemented such in a tests/game-state/test_mode.c file.

### CODE REVIEW

File: tests/game-state/test_mode.c [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    cr_assert_not_null(new_mode, "game_mode_new() test has failed!");
+
+    game_mode_free(new_mode);
+}
+
+/* Checks the initialization of a mode struct */
+Test(mode, init)
+{
+    game_mode_t *mode = game_mode_new(NORMAL, NULL, "normal");
```
Comment (jeyeong) [inline tests/game-state/test_mode.c:28]: This probably isn't a big deal, but if you are just trying to test game_mode_init(), is it necessary to use game_mode_new() here? Unless you're interested in the re-initialization of mode.

File: tests/game-state/test_mode.c [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+/* Checks that normal mode is loaded correctly */
+Test(mode, load_normal)
+{
+    game_t g;
+
+    int rc = load_normal_mode(&g);
+
+    cr_assert_eq(rc, SUCCESS, "load_normal_mode() has failed!");
+
+    cr_assert_not_null(g.mode, "load_norma_mode() has failed!");
```
Comment (jeyeong) [inline tests/game-state/test_mode.c:64]: Typo

**Review by jeyeong — APPROVED** [review]
Just two minor points. But generally looks good to go!

File: tests/game-state/test_mode.c (hunk 1 above)
Comment (pkasner) [reply tests/game-state/test_mode.c:28]: The init function specifies that it should be given an already allocated game_mode struct in mode.h. This is to ensure space is allocated for the mode_ctx, which uses strncpy().

**Review by lilyehsani — APPROVED** [review]
Looks great! Approved.

**Comment by pkasner** [comment]
These tests suffice for the very basic functions currently included in mode.h. However, as more complex functions, such as run_conversation_mode and other mode operation functions, this file need to be expanded with additional tests.

**Comment by lilyehsani** [comment]
Issue Score: ✔️++

Comments:
Great work overall!

**Pull request merged**